### ISSUE TITLE
Modifying variable from outer scope without `nonlocal` doesn't get marked as error

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#1253]
## Environment data

-   Language Server version: 2021.5.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.8 Anaconda

## Sample

```python
def outer():
    a = 1

    def inner():
        # Missing: nonlocal a
        a += 1

    inner()

    return a

outer()
```

## Expected behavior

In function `inner`, `a` should not be writable without `nonlocal a`.

## Actual behavior

No error reported.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by Azureblade3808** [comment]
@jakebailey This one has been fixed (in 2021.5.3 I think), but is not mentioned in the change log. Please close and modify the change log if needed.

**Comment by jakebailey** [comment]
Oops, we forgot to add the label we use to ensure it gets closed. Thanks for noticing!
